medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -38,21 +38,108 @@ pub async fn delete_service_worker(
     ) -> anyhow::Result<()> {
-        tracing::info!("Setting error state in state manager");
-        state_manager
-            .set_status(
-                self.message.job_id,
-                JobStatus::Error {
-                    message: e.to_string(),
-                    code: "".to_string(),
-                },
-            )
-            .await?;
-        let _ = self.remove().await?;
-        Ok(())
+        let message = self.message.clone();
+        handle_processing_error_for_delete_message(self, message, error, state_manager).await
+    }
+}
+
+async fn handle_processing_error_for_delete_message<T>(
+    removeable: T,
+    message: DeleteMessage,
+    error: anyhow::Error,
+    state_manager: &(dyn JobStatusClient + Send + Sync),
+) -> anyhow::Result<()>
+where
+    T: Removeable,
+{
+    tracing::info!("Setting error state in state manager");
+    state_manager
+        .set_status(
+            message.job_id,
+            JobStatus::Error {
+                message: error.to_string(),
+                code: "".to_string(),
+            },
+        )
+        .await?;
+    let _ = removeable.remove().await?;
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::models::{DeleteMessage, JobStatusResponse};
+    use tokio_test::block_on;
+
+    struct ShouldRemove {}
+
+    #[async_trait]
+    impl Removeable for ShouldRemove {
+        async fn remove(&self) -> Result<String, anyhow::Error> {
+            Ok("success".to_owned())
+        }
+    }
+
+    struct TestJobStatusClient {}
+
+    #[async_trait]
+    impl JobStatusClient for TestJobStatusClient {
+        async fn get_status(
+            &self,
+            _id: Uuid,
+        ) -> Result<crate::models::JobStatusResponse, crate::state_manager::MyRedisError> {
+            unimplemented!()
+        }
+        async fn set_status(
+            &self,
+            id: Uuid,
+            status: JobStatus,
+        ) -> Result<crate::models::JobStatusResponse, crate::state_manager::MyRedisError> {
+            Ok(JobStatusResponse { id, status })
+        }
+    }
+
+    fn given_an_error_has_occurred() -> anyhow::Error {
+        anyhow!("literally any error")
+    }
+
+    fn given_we_have_a_delete_message() -> DeleteMessage {
+        DeleteMessage {
+            document_content_id: "our_id".to_owned(),
+            job_id: Uuid::new_v4(),
+        }
+    }
+
+    fn when_we_handle_the_error(
+        message: DeleteMessage,
+        error: anyhow::Error,
+        state_manager: TestJobStatusClient,
+        removeable: ShouldRemove,
+    ) -> Result<(), anyhow::Error> {
+        block_on(handle_processing_error_for_delete_message(
+            removeable,
+            message,
+            error,
+            &state_manager,
+        ))
+        .map_err(|e| {
+            println!("{:#?}", e.to_string());
+            e
+        })
+    }
+
+    #[test]
+    fn test_an_error_removes_delete_message() {
```

Comment:
@majikandy I've changed our first test to describe the current behaviour. I'd like us to continue writing a suite of tests that cover current behaviour in this PR.